cypress/support/readPdf.ts

```diff
@@ -2,12 +2,25 @@ import fs from "fs";
 import path from "path";
 import pdf from "pdf-parse";
 
-const readPdf = async (pathToPdf: string): Promise<string> => {
+const delay = (time: number): Promise<void> => {
+    return new Promise((resolve) => setTimeout(resolve, time));
+};
+
+const readPdf = async (pathToPdf: string, maxRetries = 4): Promise<string> => {
     const pdfPath = path.resolve(pathToPdf);
-    const dataBuffer = fs.readFileSync(pdfPath);
 
-    const { text } = await pdf(dataBuffer);
-    return text.replaceAll("\n", "");
+    for (let retries = 0; retries < maxRetries; retries++) {
+        try {
```

Comment:
What is the retry thing for? Is this the thing you mentioned with having to run the test multiple times?

Reply:
Yes and no. Before this I was using the cypress retry which I think is too much of a hack. This one I am reading the file and if it fails then wait before retrying again until a max is reach, This is better than just using the cypress retry imo.